packages/@n8n/config/src/configs/credentials.config.ts

```diff
@@ -22,4 +22,10 @@ export class CredentialsConfig {
 
 	@Nested
 	overwrite: CredentialsOverwrite;
+
+	/**
+	 * If enabled, some credentials are allowed to use system-level credentials (e.g., ECS, EKS, ENV, etc.).
+	 */
+	@Env('CREDENTIALS_ALLOW_SYSTEMS')
+	allowSystems: boolean = false;
```

Comment:
I'm actually thinking of making it "modable", as-in take an enum of values:

- `disable`: Do not allow system credential.
- `enable-all`: Anyone can create system credential.
- `enable-admin`: Only admin users can create system credential

This way this can mixed with Enterprise sharing such that an admin can create a few system credentials as needed (in the future with `assumeRole` support the admin can create role specific credentials) and share them to specific users or project.